metals/src/main/scala/scala/meta/internal/metals/ClientExperimentalCapabilities.scala

```diff
@@ -1,58 +1,87 @@
 package scala.meta.internal.metals
-import com.google.gson.JsonElement
-import com.google.gson.JsonNull
+import com.google.gson.JsonObject
 import org.eclipse.{lsp4j => l}
 
+/**
+ * While all these values can be set here, they are here only for
+ * compatablity with clients that were setting them this way. From
+ * a client perspective it's preferable and recommended to use
+ * InitializationOptions instead. From a development perspective
+ * don't add something here unless it's truly a more "experimental"
+ * type feature.
+ */
 final case class ClientExperimentalCapabilities(
-    debuggingProvider: Boolean,
-    decorationProvider: Boolean,
-    didFocusProvider: Boolean,
-    doctorProvider: String,
-    executeClientCommandProvider: Boolean,
-    inputBoxProvider: Boolean,
-    openFilesOnRenameProvider: Boolean,
-    quickPickProvider: Boolean,
-    slowTaskProvider: Boolean,
-    statusBarProvider: String,
-    treeViewProvider: Boolean
+    debuggingProvider: Option[Boolean],
+    decorationProvider: Option[Boolean],
+    didFocusProvider: Option[Boolean],
+    doctorProvider: Option[String],
+    executeClientCommandProvider: Option[Boolean],
+    inputBoxProvider: Option[Boolean],
+    openFilesOnRenameProvider: Option[Boolean],
+    quickPickProvider: Option[Boolean],
+    slowTaskProvider: Option[Boolean],
+    statusBarProvider: Option[String],
+    treeViewProvider: Option[Boolean]
 ) {
-  def this() =
-    this(
-      debuggingProvider = false,
-      decorationProvider = false,
-      didFocusProvider = false,
-      doctorProvider = "html",
-      executeClientCommandProvider = false,
-      inputBoxProvider = false,
-      openFilesOnRenameProvider = false,
-      quickPickProvider = false,
-      slowTaskProvider = false,
-      statusBarProvider = "off",
-      treeViewProvider = false
-    )
-  def doctorFormatIsJson: Boolean = doctorProvider == "json"
-  def statusBarIsOn: Boolean = statusBarProvider == "on"
-  def statusBarIsOff: Boolean = statusBarProvider == "off"
-  def statusBarIsShowMessage: Boolean = statusBarProvider == "show-message"
-  def statusBarIsLogMessage: Boolean = statusBarProvider == "log-message"
+  def doctorFormatIsJson: Boolean = doctorProvider.exists(_ == "json")
+  def statusBarIsOn: Boolean = statusBarProvider.exists(_ == "on")
```

Comment:
if we had 'Option[Boolean]' here we could use `extract` in `ClientConfiguration.statusBarIsOn`

because right now `statusBarIsShowMessage == true` AND `statusBarIsLogMessage == true` is possible if you set one in experimental capabilities and second in initialization options
:thinking:

Reply:
> if we had `Option[Boolean]` here we could use extract in ClientConfiguration.statusBarIsOn

Well the tricky thing is that it's not a binary decision right? It can be on, off, show message, or log message. So that can't really be represented this way right? So I'm unsure how to do that.

> because right now statusBarIsShowMessage == true AND statusBarIsLogMessage == true is possible if you set one in experimental capabilities and second in initialization options

Sort of? I guess a client could set `statusBarProvider = "on"` in `InitializationOptions` and then `statusBarProvider = "log-message"` in `ClientExperimentalCapabilities`, but I _believe_ all of the checks would check of `statusBarIsOn` all of the time, which would always cause that one to win. Ideally, we are allowing illegal states to be represented by allowing for them to be set in both places. I sort of wish that we could remove the `ClientExperimentalCapabilities` since they can all be set via `InitializationOptions`. Is that allowed? If you think it's fine to potentially break a client that is being set that way, then I'm totally ok with removing them. I think if we are explicit and up front about this happening, then maybe we can? In reality, these are marked "experimental", which implies that they won't always be here.

Reply:
Alright, thanks for explaining this. After you showed me I got what you meant 😆 These changes are now in this branch.